README.md

```diff
@@ -31,23 +34,70 @@ To run in watch mode, add `-w`/`--watch` to the command:
 npx steiger ./src --watch
 ```
 
-# Configuration
+## Configuration
 
 Steiger is zero-config! If you don't want to disable certain rules, you can safely skip this section.
 
 Steiger is configurable via `cosmiconfig`. That means that you can create a `steiger.config.ts` or `steiger.config.js` file in the root of your project to configure the rules. Import `{ defineConfig } from "steiger"` to get autocompletion.
 
-```ts
-import { defineConfig } from 'steiger'
+The config file shape is highly inspired by ESLint's config file, so if you have configured ESLint before, you'll find it easy to configure Steiger.
+
+### Example
+
+Here are some rules on how configuration is processed:
 
-export default defineConfig({
-  rules: {
-    'public-api': 'off',
+- Config objects are processed from top to bottom, so if there are multiple config object that match the same file for the same rule, the last one will be applied.
+- GlobalIgnore objects `{ ignores: ['**/__mocks__/**'] }` are applied to all rules, they are processed first and permanently remove files from the linter's field of view, so you can't reassign severity to them in other later objects.
+- You can set options for a rule once. When set, options are applied for the entire file system that is covered by Steiger.
+
+Note that this line `...fsd.configs.recommended,` just takes the plugin and the recommended rules configuration (all enabled with "error" severity by default) and puts it into the config array.
+
+```javascript
+// ./steiger.config.ts
+import fsd from '@feature-sliced/steiger-plugin'
+import defineConfig from 'steiger'
+
+export default defineConfig([
+  ...fsd.configs.recommended,
+  {
+    // ignore all mock files for all rules
+    ignores: ['**/__mocks__/**'],
+  },
+  {
+    files: ['./src/shared/**'],
+    rules: {
+      // disable public-api rule for files in /shared folder
+      'fsd/public-api': 'off',
+    },
   },
-})
+  {
+    files: ['./src/widgets/**'],
+    ignores: ['**/discount-offers/**'],
+    rules: {
+      // disable no-segmentless-slices rule for all widgets except /discount-offers
+      'fsd/no-segmentless-slices': 'off',
+    },
+  },
+])
 ```
 
-# Rules
+### Glob matching
+
+All globs are matched only against files, folder severities are computed based on the files inside them. The formula is simple: the folder severity is the highest severity of files inside it (from highest to lowest: error, warn, off).
+
+**Glob examples**:
+
+- `./src/shared/**` - matches all files in the `shared` folder and its subfolders
+- `./src/shared/*` - matches all files that are direct children of the `shared` folder
+- `./src/shared` - based on the fact that globs are matched against files, this one matches only `shared` file (without an extension) inside the `src` folder
+- `**/__mocks__/**` - matches all files in all `__mocks__` folders throughout the project
+- `**/*.{test,spec}.{ts,tsx}` - matches all test files in the project
```

Comment:
issue: again, I believe this is a more technical description, and I forecast that most people will just disable entire folders, so people are unlikely to run up against a case where one file made the whole folder be `error`.

suggestion: let's just get rid of this entire section

Reply:
Decided to move it to CONFIG_EXAMPLES.md so far. Let me know if you think it should not be even there

Reply:
I personally think it shouldn't be even there, but I also don't mind it being there. As long as it's not up front in the README, I'm fine with it